File: example/src/main/java/ru/virgil/example/stats/StatsController.java
package ru.virgil.example.stats;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import ru.virgil.example.box.BoxService;
import ru.virgil.example.order.BuyingOrderService;
import ru.virgil.example.truck.TruckCounterModule;
import ru.virgil.example.truck.TruckService;
import ru.virgil.example.user.UserDetails;
import ru.virgil.example.user.UserDetailsService;

@RestController
@RequestMapping("/stats")
@RequiredArgsConstructor
public class StatsController {

    private final BoxService boxService;
    private final TruckService truckService;
    private final TruckCounterModule truckCounterModule;
    private final BuyingOrderService buyingOrderService;
    private final UserDetailsService userDetailsService;

    @GetMapping("/all")
    public StatsDto getAllStats() {
        long boxesCount = boxService.getRepository().count();
        long trucksCount = truckService.getRepository().count();
        long ordersCount = buyingOrderService.getRepository().count();
        return new StatsDto(boxesCount, trucksCount, ordersCount);
    }

    @GetMapping("/my")
    public StatsDto getMyStats() {
        UserDetails owner = userDetailsService.getCurrentUser();
        long boxesCount = boxService.countMyBoxes(owner);
        long trucksCount = truckCounterModule.countAll(owner);
        long ordersCount = buyingOrderService.countMy(owner);
        return new StatsDto(boxesCount, trucksCount, ordersCount);
    }
}

File: example/src/main/java/ru/virgil/example/stats/StatsController.kt
package ru.virgil.example.stats

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import ru.virgil.example.box.BoxRepository
import ru.virgil.example.box.BoxService
import ru.virgil.example.order.BuyingOrderRepository
import ru.virgil.example.order.BuyingOrderService
import ru.virgil.example.truck.TruckRepository


@RestController
@RequestMapping("/stats")
class StatsController(
    private val boxService: BoxService,
    private val boxRepository: BoxRepository,
    private val truckRepository: TruckRepository,
    private val buyingOrderRepository: BuyingOrderRepository,
    private val buyingOrderService: BuyingOrderService,
) {

    @GetMapping("/all")
    fun getAllStats(): StatsDto {
        val boxesCount = boxRepository.count()
        val trucksCount = truckRepository.count()
        val ordersCount = buyingOrderRepository.count()
        return StatsDto(boxesCount, trucksCount, ordersCount)
    }

    @GetMapping("/my")
    fun getMyStats(): StatsDto {
        val boxesCount = boxService.countMy()
        val trucksCount = truckRepository.count()
        val ordersCount = buyingOrderService.countMy()
        return StatsDto(boxesCount, trucksCount, ordersCount)
    }
}
